src/mrsimulator/utils/error.py

```diff
@@ -27,7 +27,38 @@ class ImmutableEventError(Exception):
     def __init__(self, cls_name, message=None):
         message = message or (
             f"Event objects are immutable for {cls_name} class. Remove all event "
-            "objects or use `Method1D`/`Method2D` class."
+            "objects or use the generic `Method` class."
+        )
+        super().__init__(message)
+
+
+class MixedSpectralDimensionTypeError(Exception):
+    """Exception raised when SpectralDimension objects and dictionaries are used in the
+    same spectral_dimension list
+
+    Attributes:
+        message: explanation of the error.
+    """
+
+    def __init__(self, message=None):
+        message = message or (
+            "Both dict and SpectralDimension objects found in spectral dimension list."
```

Comment:
Add a space after `list.` Always end multi-line strings with a space. Python adds the strings without adding space in between.